### PR TITLE
DOCKER DEV ENV

### PR DESCRIPTION
**Pull request by amoffat**
Hopefully this helps others iterate quicker on their code with fewer errors.  See the [docker readme](https://github.com/amoffat/Chaos/blob/docker-dev-env/dev/docker/README.md)

It should make things as simple as going `./run.sh` and having a full chaos server, in a second, that looks at your own personal chaos fork.  This way you can test things like PRs, labels, etc, with full control on your fork, before opening a PR here.

### CODE REVIEW

**Comment by hongaar** [comment]
Awesome! Can't wait to try it..

File: requirements.txt [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
 cryptography==1.8.1
 packaging==16.8
 pyparsing==2.2.0
 python-dateutil==2.6.0
 requests==2.14.2
 schedule==0.4.2
+sh==1.12.13
 six==1.10.0
-emoji==0.4.5
+Werkzeug==0.12.2
```
Comment (sorcio) [inline requirements.txt]: What are `click` and `Flask` used for?

Comment (amoffat) [reply requirements.txt]: good eye.  i was testing some stuff and :hankey: 'd my local venv requirements.  let me fix..

**Review by sorcio — APPROVED** [review]

File: dev/docker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+WORKDIR $chaosdir
+
+COPY requirements.txt .
+RUN pip install -r requirements.txt
+RUN rm requirements.txt
+
+RUN rm /etc/nginx/sites-enabled/default
+
+RUN printf "\
```
Comment (xyproto) [inline dev/docker/Dockerfile:56]: If `#!/bin/bash` is added here....

File: dev/docker/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    ln -sf /root/workspace/Chaos/etc/nginx/chaos_errors /etc/nginx/sites-available/\n\
+    ln -sf /etc/nginx/sites-available/chaos_errors /etc/nginx/sites-enabled/\n\
+
+    service supervisor start\n\
+    service nginx start\n\
+
+    sleep 1\n\
+
+    tail -F /var/log/supervisor/chaos-*"\
+    >> /root/start_chaos.sh
```
Comment (xyproto) [inline dev/docker/Dockerfile:67]: ...and this one is piped to `/usr/bin/start_chaos` instead, followed by a `chmod +x /usr/bin/start_chaos`...

File: dev/docker/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+EXPOSE 80 8081
+ENTRYPOINT bash /root/start_chaos.sh
```
Comment (xyproto) [inline dev/docker/Dockerfile:70]: ... then this line could just be `ENTRYPOINT start_chaos`?

File: dev/vagrant/README.md [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+To run your own VM for local development, install [Vagrant](https://www.vagrantup.com/downloads.html) and [VirtualBox](https://www.virtualbox.org/wiki/Downloads). Then, type on your host machine:
+
+    vagrant up
+    vagrant ssh
+
+When in SSH:
+
+    sudo su
+    cd /vagrant
+    python3 chaos.py
```
Comment (xyproto) [inline dev/vagrant/README.md:12]: why not `sudo python3 chaos.py` instead of `sudo su`?

Comment (amoffat) [reply dev/vagrant/README.md:12]: The vagrant README.md came from the top-level README.md without changes.  I agree we should clean it up though once this lands

File: dev/docker/Dockerfile (hunk 4 above)
Comment (amoffat) [reply dev/docker/Dockerfile:70]: Thanks @xyproto, these are good improvements.  I'll leave that for a followup PR so as to not extend voting on this 12 hour(!) PR

Comment (xyproto) [reply dev/docker/Dockerfile:70]: Understandable.

**Pull request merged**